lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
```

Comment:
I think we care more about `valid?` as a command here than its actual return value, so I'm going to side with Derek on keeping these separate. I do, however, think that it makes more sense to use `empty?` instead of `blank?`.